shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-mysql/src/main/java/org/apache/shardingsphere/sql/parser/mysql/visitor/impl/MySQLDMLVisitor.java

```diff
@@ -470,7 +470,8 @@ private ASTNode createProjection(final ProjectionContext ctx, final AliasSegment
         }
         if (projection instanceof CommonExpressionSegment) {
             CommonExpressionSegment segment = (CommonExpressionSegment) projection;
-            ExpressionProjectionSegment result = new ExpressionProjectionSegment(segment.getStartIndex(), segment.getStopIndex(), segment.getText());
+            ExpressionProjectionSegment result = null == alias ? new ExpressionProjectionSegment(segment.getStartIndex(), segment.getStopIndex(), segment.getText())
+                    : new ExpressionProjectionSegment(segment.getStartIndex(), alias.getStopIndex(), segment.getText());
```

Comment:
Do you think it is better to overwrite `getStopIndex()` of `ExpressionProjectionSegment ` with the stopIndex of its`aliasSegment`,

Reply:
> Do you think it is better to overwrite `getStopIndex()` of `ExpressionProjectionSegment ` with the stopIndex of its`aliasSegment`,

Ok

Reply:
> > Do you think it is better to overwrite `getStopIndex()` of `ExpressionProjectionSegment ` with the stopIndex of its`aliasSegment`,
> 
> I don't think so because expressionProjectionSegment contains alias.